console/backend/src/main/java/org/frankframework/management/web/TestPipeline.java

```diff
@@ -150,30 +150,17 @@ private String processZipFile(InputStream file, RequestMessageBuilder builder) t
 		StringBuilder result = new StringBuilder();
 
 		ZipInputStream archive = new ZipInputStream(file);
-		for (ZipEntry entry=archive.getNextEntry(); entry!=null; entry=archive.getNextEntry()) {
-			String name = entry.getName();
-			int size = (int)entry.getSize();
-			if (size>0) {
-				byte[] b=new byte[size];
-				int rb=0;
-				int chunk;
-				while ((size - rb) > 0) {
-					chunk=archive.read(b,rb,size - rb);
-					if (chunk==-1) {
-						break;
-					}
-					rb+=chunk;
-				}
-				String currentMessage = XmlEncodingUtils.readXml(b, null);
-
-				builder.setPayload(currentMessage);
-				Message<?> response = sendSyncMessage(builder);
-				result.append(name);
-				result.append(": ");
-				result.append(BusMessageUtils.getHeader(response, ResponseMessageBase.STATE_KEY));
-				result.append("\n");
-			}
-			archive.closeEntry();
+		ZipEntry zipEntry;
+		while ((zipEntry = archive.getNextEntry()) != null) {
+			String name = zipEntry.getName();
+			String currentMessage = XmlEncodingUtils.readXml(StreamUtil.streamToBytes(StreamUtil.dontClose(archive)), null);
```

Comment:
Is this the easiest way to read from a stream into a String?
What about e.g. `String text = new String(inputStream.readAllBytes());`
or `String currentMessage = XmlEncodingUtils.readXml(archive, "UTF-8");`

Reply:
You must not close the stream!